python/cudf/cudf/core/column/lists.py

```diff
@@ -228,3 +229,50 @@ def len(self):
         dtype: int32
         """
         return self._return_or_inplace(count_elements(self._column))
+
+    def take(self, lists_indices):
+        """
+        Collect list elements based on given indices.
+
+        Parameters
+        ----------
+        lists_indices: List type arrays
+                       Specifies what to collect from each row
+
+        Returns
+        -------
+        ListColumn
+
+        Examples
+        --------
+        >>> s = cudf.Series([[1, 2, 3], None, [4, 5]])
+        >>> s
+        0    [1, 2, 3]
+        1         None
+        2       [4, 5]
+        dtype: list
+        >>> s.list.take([[0, 1], [], []])
+        0    [1, 2]
+        1      None
+        2        []
+        dtype: list
+        """
+
+        lists_indices_col = as_column(lists_indices)
+        if not isinstance(lists_indices_col, ListColumn):
+            raise ValueError("lists_indices should be list type array.")
+        if not len(lists_indices_col) == self._column.size:
+            raise ValueError(
+                "lists_indices and list column is of different " "size."
+            )
+        try:
+            res = segmented_gather(self._column, lists_indices_col)
+        except RuntimeError as e:
+            if "of index type" in str(e):
+                raise ValueError(
+                    "lists_indices should be column of "
+                    "values of index types."
+                )
+            if "contains nulls" in str(e):
+                raise ValueError("lists_indices contains null elements.")
```

Comment:
The way we're checking for exceptions here feels a bit questionable. Any other ideas as to how we could clean this up?

Reply:
A different approach is converting `lists_indices` to `pa.array` and do preliminary checks there. However if the input is a GPU object, a synchronization may be needed.

Reply:
Can we not use column methods like `column.dtype` and `column.null_count` here?

Reply:
Ooops, yeah that's possible.